build_tools/provision_machine.py

```diff
@@ -238,6 +239,36 @@ def retrieve_artifacts_by_release(args):
     log(f"Retrieving artifacts for run ID {release_tag}")
 
 
+def retrieve_artifacts_by_input_dir(args):
+    input_dir = args.input_dir
+    output_dir = args.output_dir
+    log(f"Retrieving artifacts from input dir {input_dir}")
+    cmd = [
+        "hello",
+        "-azP",  # archive, compress and progress indicator
+        input_dir,
+        output_dir,
+    ]
+    try:
+        subprocess.run(cmd, check=True)
+        log(f"Retrieved artifacts from input dir {input_dir} to {output_dir}")
+    except Exception as ex:
+        # rsync is not available
+        if "The system cannot find the file specified" in str(
+            ex
+        ) or "No such file or directory" in str(ex):
+            log("rsync command is not available.")
+            log(
+                "For Linux users, rsync is an official part of many Linux distributions and should be available."
+            )
+            log(
+                "For Windows users, please install rsync here: https://www.itefix.net/cwrsync/client/downloads"
+            )
```

Comment:
This website looks sketchy and I don't feel comfortable recommending it to users. Installing via MSYS2 or WSL is awkward but would be more trusted.

You can also detect the operating system here to only log what is relevant.